src/System.Net.Sockets/src/System/Net/Sockets/SocketPal.Unix.cs

```diff
@@ -981,6 +989,21 @@ public static unsafe SocketError GetSockOpt(SafeCloseSocket handle, SocketOption
                 optLen = 0;
                 err = Interop.Sys.GetSockOpt(handle, optionLevel, optionName, null, &optLen);
             }
+            else if (optionName == SocketOptionName.Error && optionValue.Length >= sizeof(int))
+            {
+                int outError;
+                SocketError returnError = GetSockOpt(handle, optionLevel, optionName, out outError);
```

Comment:
This is calling the other method, which calls GetSocketErrorOption.  I was trying to keep the logic around calling GetSocketErrorOption and doing the translation consolidated in one place.  This method then takes that result and puts the int into the byte[] array.